plugins/modules/katello_repository.py

```diff
@@ -88,10 +83,41 @@
       - on_demand
     required: false
     type: str
-  gpg_key:
+  mirror_on_sync:
     description:
-    - Repository GPG key
+      - toggle "mirror on sync" where the state of the repository mirrors that of the upstream repository at sync time
+    default: true
+    type: bool
     required: false
+  upstream_username:
+    description:
+      - username to access upstream repository
+    type: str
+  upstream_password:
+    description:
+      - password to access upstream repository
+    type: str
+  docker_upstream_name:
+    description:
+      - name of the upstream docker repository
+      - only available for I(content_type)==I(docker)
```

Comment:
the [examples](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#linking-within-module-documentation) use `I(content_type=docker)`
```suggestion
      - only available for I(content_type=docker)
```